### PR TITLE
Switch to id_from_acronym generator

### PR DESCRIPTION
**Pull request by lecriste**
**Issue:**
Region ids are not guaranteed to always remain unchanged when splitting for v2 and v3 annotations and their versions.

**Requirements:**
* Unique ids across the hierarchy
* Deterministic mapping from acronym to id
* Stable ids across v2 and v3 annotations and versions

**Proposal:**
* For existing acronyms return hierarchy id
* For new acronyms return a sha256 digest from the acronym truncated at 10 digits

**Use-case:**
https://docs.google.com/presentation/d/1shlysZb4ZCTH88nCWoaDCzgC-5ZtshVaTe0OtUIyfCo/edit?usp=sharing

### CODE REVIEW

**Comment by eleftherioszisis** [comment]
Is there a ticket about this change? If not could you please provide some motivation for this change? Why is it needed? What issue does it solve? etc.
Thanks!

**Comment by lecriste** [comment]
There is no ticket, I added a link to a couple of slides in the description.

File: atlas_splitter/utils.py [hunk 1]
```diff
@@ -46,6 +47,27 @@ def create_id_generator(region_map: RegionMap) -> Iterator[int]:
...
+    Return:
+        region id.
+    """
+    region_id_set = region_map.find(acronym, attr="acronym", with_descendants=False)
+    if region_id_set:
+        [region_id] = region_id_set
+    else:  # acronym not present in hierarchy, generating a corresponding id
+        sha = hashlib.sha256()
+        sha.update(acronym.encode("utf-8"))
+        region_id = int(str(int(sha.hexdigest(), 16))[0:10])
```
Comment (eleftherioszisis) [inline atlas_splitter/utils.py:66]: You can reduce the integer hash with a modulo operation:
```python
sha = hashlib.sha256(acronym.encode("utf-8"))
region_id = int(sha.hexdigest(), 16) % 10 ** 10
```

File: atlas_splitter/layer_splitter/isocortex_layer_23.py [hunk 2]
```diff
@@ -127,23 +125,24 @@ def edit_hierarchy(
             # Create children
             new_children = []
             for layer in ["layer_2", "layer_3"]:
+                new_child = copy.deepcopy(child)
+                new_child["acronym"] = child["acronym"][:-3] + layer[-1]
```
Comment (eleftherioszisis) [inline atlas_splitter/layer_splitter/isocortex_layer_23.py:129]: How does this work for layers that are not ending to integers?

Comment (lecriste) [reply atlas_splitter/layer_splitter/isocortex_layer_23.py:129]: I don't know, I did not change this:
```
- new_child["acronym"] = child["acronym"][:-3]
- new_child["acronym"] = new_child["acronym"] + layer[-1]
+ new_child["acronym"] = child["acronym"][:-3] + layer[-1]
```

File: atlas_splitter/utils.py (hunk 1 above)
Comment (lecriste) [reply atlas_splitter/utils.py:66]: Yes but if `int(sha.hexdigest(), 16) % 10` = 0, then `region_id` stays 0...

Comment (eleftherioszisis) [reply atlas_splitter/utils.py:66]: Yeah, indeed. ok leave it as it is

File: atlas_splitter/layer_splitter/isocortex_layer_23.py (hunk 2 above)
Comment (eleftherioszisis) [reply atlas_splitter/layer_splitter/isocortex_layer_23.py:129]: I assume given that only 2/3 layer is mentioned in this module, that this is the case for the splitter. So we are good.

Comment (eleftherioszisis) [reply atlas_splitter/layer_splitter/isocortex_layer_23.py:129]: I missed the line just above `if child["acronym"].endswith("2/3"):`
It's good indeed.

**Review by eleftherioszisis — APPROVED** [review]

**Pull request merged**